system-int-test/test/cmr/system_int_test/search/granule_shapefile_search_test.clj

```diff
@@ -32,6 +33,99 @@
+      (are3 [shapefile additional-params regex]
+        (is (re-find regex
+                      (first (:errors (search/find-refs-with-multi-part-form-post
+                                        :granule
+                                        (let [params [{:name "shapefile"
+                                                       :content (io/file (io/resource (str "shapefiles/" shapefile)))
+                                                       :mime-type "application/shapefile+zip"}]]
+                                          (if (seq additional-params)
+                                            (conj params additional-params)
+                                            params)))))))
+                                  
+        "All granules query"
+        "box.zip" {} #"The CMR does not allow querying across granules in all collections with a spatial condition"
+
+        "Corrupt zip file"
+        "corrupt_file.zip" {:name "provider" :content "PROV1"} #"Error while uncompressing zip file: invalid END header \(bad central directory offset\)"
+        
+        "Missing .shp file"
+        "missing_shapefile_shp.zip" {:name "provider" :content "PROV1"} #"Incomplete shapefile: missing .shp file"
+
+        "Shapefile is too big"
+        "too_big.zip" {:name "provider" :content "PROV1"} #"Shapefile size exceeds the 50000 byte limit"
+
+        "Shapefile has too many features"
+        "too_many_features.zip" {:name "provider" :content "PROV1"} #"Shapefile feature count \[3\] exceeds the 2 feature limit"
+
+        "Shapefile has too many points"
+        "too_many_points.zip" {:name "provider" :content "PROV1"} #"Number of points in shapefile exceeds the limit of 50"))
+
+    (testing "GeoJSON Failure cases"
+      (are3 [shapefile additional-params regex]
+        (is (re-find regex
+                      (first (:errors (search/find-refs-with-multi-part-form-post
+                                        :granule
+                                        (let [params [{:name "shapefile"
+                                                       :content (io/file (io/resource (str "shapefiles/" shapefile)))
+                                                       :mime-type "application/geo+json"}]]
+                                          (if (seq additional-params)
+                                            (conj params additional-params)
+                                            params)))))))
+                                  
+        "All granules query"
+        "polygon_with_hole.geojson" {} #"The CMR does not allow querying across granules in all collections with a spatial condition"
+
+        "Failed to parse GeoJSON file"
+        "invalid_json.geojson" {:name "provider" :content "PROV1"} #"Failed to parse GeoJSON file"
+
+        "Shapefile has too many features"
+        "too_many_features.geojson" {:name "provider" :content "PROV1"} #"GeoJSON feature count \[3\] exceeds the 2 feature limit"
```

Comment:
Just a comment: I find the file name and all the variable names confusing. I wish this file was named "three_features.geojson" and so did everything related to `too_many_features`. It is only too many when you set the maximum features to `2`.

Reply:
I think it's more important that the names identify what they are used for. 'three_features.geojson' doesn't indicate what it is for. The fact that it has three features is not important.